src/execution/__tests__/union-interface-test.ts

```diff
@@ -154,6 +154,77 @@ odie.mother.progeny = [odie];
 const liz = new Person('Liz');
 const john = new Person('John', [garfield, odie], [liz, odie]);
 
+const SearchableInterface = new GraphQLInterfaceType({
+  name: 'Searchable',
+  fields: {
+    id: { type: GraphQLString },
+  },
+});
+
+const TypeA = new GraphQLObjectType({
+  name: 'TypeA',
+  interfaces: [SearchableInterface],
+  fields: () => ({
+    id: { type: GraphQLString },
+    nameA: { type: GraphQLString },
+  }),
+  isTypeOf: (_value, _context, _info) => {
+    return new Promise((_resolve, reject) => {
+      setTimeout(() => {
+        reject(new Error('TypeA_isTypeOf_rejected'));
+      }, 10);
+    });
```

Comment:
I think the lint settings would rather you wrote this without braces:

```suggestion
  isTypeOf: (_value, _context, _info) =>
    new Promise(
      (_resolve, reject) =>
        void setTimeout(
          () => void reject(new Error("TypeA_isTypeOf_rejected")),
          10,
        ),
    ),
```

But this is just stylistic guidance so that all code looks as self-similar as possible, to make it easier on readers/reviewers. As the maintainer you're completely free to change these rules. I personally think it's fine to have arrow functions with block bodies; but I also think the `void` keyword should be used more :)